### PR TITLE
Adding vsftp template

### PR DESCRIPTION
**Pull request by ramamaddodi**
Added init.pp and vsftp template

Initial commit

Removed empty.conf.erb template

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -26,25 +26,31 @@
...
   $connect_from_port_20    = 'YES',
+  $connect_timeout         = '60',
   $chown_uploads           = 'NO',
   $chown_username          = undef,
   $xferlog_file            = '/var/log/vsftpd.log',
   $xferlog_std_format      = 'YES',
   $idle_session_timeout    = '600',
   $data_connection_timeout = '120',
-  $nopriv_user             = undef,
+  $nopriv_user             = 'ftpsecure',
```
Comment (ghoneycutt) [inline manifests/init.pp:39]: for values that already exist, we should just set those in Hiera. We do want to submit this PR back to upstream.

File: templates/empty.conf.erb [hunk 2]
```diff
@@ -1,9 +0,0 @@
-# This is vsftpd.conf created from scratch by puppet.
```
Comment (ghoneycutt) [inline templates/empty.conf.erb:1]: I don't see why this needs to be removed

File: templates/vsftpd.conf.erb [hunk 3]
```diff
@@ -24,82 +24,51 @@ local_umask=<%= @local_umask %>
 # Uncomment this to allow the anonymous FTP user to upload files. This only
 # has an effect if the above global write enable is activated. Also, you will
 # obviously need to create a directory writable by the FTP user.
-<% if @anon_upload_enable == 'YES' -%>
```
Comment (ghoneycutt) [inline templates/vsftpd.conf.erb]: why would this change?

File: templates/vsftpd.conf.erb [hunk 4]
```diff
@@ -24,82 +24,51 @@ local_umask=<%= @local_umask %>
...
-anon_upload_enable=YES
-<% else -%>
-#anon_upload_enable=YES
-<% end -%>
+#anon_upload_enable=<%= @anon_upload_enable %>
 #
 # Uncomment this if you want the anonymous FTP user to be able to create
 # new directories.
-<% if @anon_mkdir_write_enable == 'YES' -%>
```
Comment (ghoneycutt) [inline templates/vsftpd.conf.erb]: why change?

File: templates/vsftpd.conf.erb [hunk 5]
```diff
@@ -24,82 +24,51 @@ local_umask=<%= @local_umask %>
...
-anon_mkdir_write_enable=YES
-<% else -%>
```
Comment (ghoneycutt) [inline templates/vsftpd.conf.erb]: why change?

File: templates/vsftpd.conf.erb [hunk 6]
```diff
@@ -24,82 +24,51 @@ local_umask=<%= @local_umask %>
...
-#anon_mkdir_write_enable=YES
-<% end -%>
+#anon_mkdir_write_enable=<%= @anon_mkdir_write_enable %>
 #
 # Activate directory messages - messages given to remote users when they
 # go into a certain directory.
 dirmessage_enable=<%= @dirmessage_enable %>
 #
-# Activate logging of uploads/downloads.
```
Comment (ghoneycutt) [inline templates/vsftpd.conf.erb:45]: this comment should stay

File: templates/vsftpd.conf.erb [hunk 7]
```diff
@@ -24,82 +24,51 @@ local_umask=<%= @local_umask %>
...
+# The target log file can be vsftpd_log_file or xferlog_file.
+# This depends on setting xferlog_std_format parameter
 xferlog_enable=<%= @xferlog_enable %>
 #
 # Make sure PORT transfer connections originate from port 20 (ftp-data).
-connect_from_port_20=<%= @connect_from_port_20 %>
+#connect_from_port_20=<%= @connect_from_port_20 %>
```
Comment (ghoneycutt) [inline templates/vsftpd.conf.erb]: instead of commenting out, this should be conditional

File: templates/vsftpd.conf.erb [hunk 8]
```diff
@@ -24,82 +24,51 @@ local_umask=<%= @local_umask %>
...
-#
-# You may override where the log file goes if you like. The default is shown
-# below.
-xferlog_file=<%= @xferlog_file %>
-#
-# If you want, you can have your log file in standard ftpd xferlog format.
-# Note that the default log file location is /var/log/xferlog in this case.
-xferlog_std_format=<%= @xferlog_std_format %>
+#chown_uploads=<%= @chown_uploads %>
+#chown_username=<%= @chown_uploads %>
```
Comment (ghoneycutt) [inline templates/vsftpd.conf.erb]: make conditional

File: templates/vsftpd.conf.erb [hunk 9]
```diff
@@ -24,82 +24,51 @@ local_umask=<%= @local_umask %>
...
 #
 # If you want, you can arrange for uploaded anonymous files to be owned by
 # a different user. Note! Using "root" for uploaded files is not
 # recommended!
-<% if @chown_uploads == 'YES' -%>
-chown_uploads=YES
-<% else -%>
-#chown_uploads=YES
-<% end -%>
-<% if @chown_username -%>
```
Comment (ghoneycutt) [inline templates/vsftpd.conf.erb]: this should stay

File: templates/vsftpd.conf.orig [hunk 10]
```diff
@@ -1,118 +0,0 @@
-# Example config file /etc/vsftpd/vsftpd.conf
-#
```
Comment (ghoneycutt) [inline templates/vsftpd.conf.orig:2]: before this is removed, ensure we have parity within our template

**Review by ghoneycutt — CHANGES_REQUESTED** [review]

File: manifests/init.pp (hunk 1 above)
Comment (ramamaddodi) [reply manifests/init.pp:39]: I will move this value to hiera

File: templates/empty.conf.erb (hunk 2 above)
Comment (ramamaddodi) [reply templates/empty.conf.erb:1]: Not sure why we need empty template when its not used

File: templates/vsftpd.conf.erb (hunk 3 above)
Comment (ramamaddodi) [reply templates/vsftpd.conf.erb]: Mentioned change is according to what we have on current ftp/sftp production server.

File: templates/vsftpd.conf.erb (hunk 4 above)
Comment (ramamaddodi) [reply templates/vsftpd.conf.erb]: Mentioned change is according to what we have on current ftp/sftp production server.

File: templates/vsftpd.conf.orig (hunk 10 above)
Comment (ramamaddodi) [reply templates/vsftpd.conf.orig:2]: Our current template is according to our current production ftp/sftp config setting

File: templates/vsftpd.conf.erb (hunk 9 above)
Comment (ramamaddodi) [reply templates/vsftpd.conf.erb]: Again mentioned changes are as per what we have on current production template for sftp/ftp environment.

File: templates/vsftpd.conf.erb (hunk 8 above)
Comment (ramamaddodi) [reply templates/vsftpd.conf.erb]: Again mentioned changes are as per what we have on current production template for sftp/ftp environment.

File: templates/vsftpd.conf.erb (hunk 7 above)
Comment (ramamaddodi) [reply templates/vsftpd.conf.erb]: Again mentioned changes are as per what we have on current production template for sftp/ftp environment.

File: templates/vsftpd.conf.erb (hunk 6 above)
Comment (ramamaddodi) [reply templates/vsftpd.conf.erb:45]: Mentioned changes are as per what we have on current production template for sftp/ftp environment.

File: templates/vsftpd.conf.erb (hunk 5 above)
Comment (ramamaddodi) [reply templates/vsftpd.conf.erb]: Mentioned changes are as per what we have on current production template for sftp/ftp environment.

**Review by ramamaddodi — COMMENTED** [review]
Changes to template is as per current production template that we have for vsftp module

File: manifests/init.pp [hunk 11]
```diff
@@ -36,47 +38,51 @@
...
+  $one_process_model       = 'NO',
   $ascii_upload_enable     = 'NO',
   $ascii_download_enable   = 'NO',
   $ftpd_banner             = undef,
   $chroot_local_user       = 'NO',
   $chroot_list_enable      = 'NO',
   $chroot_list_file        = '/etc/vsftpd/chroot_list',
-  $ls_recurse_enable       = 'NO',
+  $secure_chroot_dir       = undef,
+  $ls_recurse_enable       = 'YES',
```
Comment (ghoneycutt) [inline manifests/init.pp]: we should not have changed the default value, instead should specify our value in hiera

File: manifests/init.pp [hunk 12]
```diff
@@ -36,47 +38,51 @@
...
+  $log_ftp_protocol        = 'NO',
   $pam_service_name        = 'vsftpd',
   $userlist_enable         = 'YES',
   $userlist_deny           = undef,
   $tcp_wrappers            = 'YES',
   $hide_file               = undef,
   $hide_ids                = 'NO',
-  $setproctitle_enable     = 'NO',
-  $text_userdb_names       = 'NO',
+  $setproctitle_enable     = undef,
```
Comment (ghoneycutt) [inline manifests/init.pp]: same here with regards to defaults and using hiera. You can specify nil which should equate to undef like this

```yaml
vsftpd::setproctitle_enable: ~
```

File: templates/vsftpd.conf.erb (hunk 7 above)
Comment (ghoneycutt) [reply templates/vsftpd.conf.erb]: do not comment out options, instead make them conditional. So if connect_from_port_20 is undef, then the configuration option is skipped.

File: templates/vsftpd.conf.erb [hunk 13]
```diff
@@ -25,80 +25,81 @@ local_umask=<%= @local_umask %>
...
 # Make sure PORT transfer connections originate from port 20 (ftp-data).
-connect_from_port_20=<%= @connect_from_port_20 %>
+#connect_from_port_20=<%= @connect_from_port_20 %> 
 #
 # If you want, you can arrange for uploaded anonymous files to be owned by
 # a different user. Note! Using "root" for uploaded files is not
 # recommended!
 <% if @chown_uploads == 'YES' -%>
-chown_uploads=YES
+chown_uploads=YES
```
Comment (ghoneycutt) [inline templates/vsftpd.conf.erb]: this is just adding trailing whitespace

**Review by ghoneycutt — CHANGES_REQUESTED** [review]

**Pull request closed without merging**